catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventToXmlElementConverter.java

```diff
@@ -99,7 +104,7 @@ private SaxEventToXmlElementConverter startConstructingElement(String uri, Strin
          * Use the uri to look up the namespace prefix and append it and the localName to the start tag
          */
         out.writeStartElement(scopedNamespaces.get(uri), localName, uri);
-        if (!checkNamespaceAdded(uri)) {
+        if (!(checkNamespaceAdded(uri) && scopeOfNamespacesAdded.peek().get(uri).contains(scopedNamespaces.get(uri)))) {
```

Comment:
It looks like you're making this same extended check in a couple of different places. Perhaps it should be refactored down into `checkNamespaceAdded` or a wrapping method instead?